R/runSearch.R

```diff
@@ -475,18 +490,41 @@ runSearch <- function(keywords,
       ) %>%
       dplyr::distinct()
 
+    candidateCodesOrig <- candidateCodes  #  temp save the list
+
+    #if domain = "drug", add drug_strength information
+
+    if (domains=="drug"){
+    candidateCodes <- candidateCodes %>%
+      dplyr::left_join(drugStrength %>%
+                         dplyr::rename("concept_id"="drug_concept_id") %>%
+                         dplyr::select(-"valid_start_date", -"valid_end_date", -"invalid_reason", -"domain_id", -"standard_concept"),
+                       by = "concept_id")
+
+    candidateCodes <- candidateCodes %>%
+      dplyr::select(
+        "concept_id", "concept_name",
+        "domain_id", "concept_class_id",
+        "vocabulary_id", "found_from",
+        "ingredient_concept_id", "amount_value", "amount_unit_concept_id",
+        "numerator_value", "numerator_unit_concept_id",
+        "denominator_value", "denominator_unit_concept_id", "box_size"
+      )
+    } else{
     candidateCodes <- candidateCodes %>%
       dplyr::select(
         "concept_id", "concept_name",
         "domain_id", "concept_class_id",
         "vocabulary_id", "found_from"
       )
+    }
 
     candidateCodes <- candidateCodes %>%
       dplyr::distinct()
 
     # remove duplicates (found in different ways)
     # keep first time it was found
+    # for drug,  same concept_id with different ingredient_concept_id will be removed as well.
```

Comment:
not sure I follow - what happens here now?

Reply:
in the drug_strength table, one drug_concept_id can have multiple lines with different ingredient_concept_id, so if we don't necessary need this information this doesn't matter .